### ISSUE TITLE
CRITICAL: A critical error has occurred trying to use the machine with name xp during 

### ISSUE THREAD

**Issue by yahyeabukar** [cuckoosandbox/cuckoo#2037]
i successfully installed cuckoo and i set up guest machine following cuckoo documentation, but i have these error. i configured my virtualbox.conf .please any help 

2018-01-01 12:03:08,436 [lib.cuckoo.core.scheduler] ERROR: Machinery error: VBoxManage exited with error restoring the machine's snapshot
2018-01-01 12:03:08,511 [lib.cuckoo.core.scheduler] CRITICAL: A critical error has occurred trying to use the machine with name xp during an analysis due to which it is no longer in a working state, please report this issue and all of the related environment details to the developers so we can improve this situation. (Note that before we would simply remove this VM from doing any more analyses, but as all the VMs will eventually be depleted that way, hopefully we'll find a better solution now).
2018-01-01 12:03:08,606 [modules.machinery.virtualbox] INFO: Successfully generated memory dump for virtual machine with label xp to path /home/codka/Downloads/cuckoo/storage/analyses/19/memory.dmp
2018-01-01 12:03:10,181 [modules.processing.behavior] WARNING: Analysis results folder does not exist at path '/home/codka/Downloads/cuckoo/storage/analyses/19/logs'.
2018-01-01 12:03:10,184 [modules.processing.memory] ERROR: Memory dump not found: to run volatility you have to enable memory_dump
2018-01-01 12:03:10,727 [modules.processing.network] ERROR: Unable to open /home/codka/Downloads/cuckoo/storage/analyses/19/dump_sorted.pcap
2018-01-01 12:03:14,637 [lib.cuckoo.core.scheduler] INFO: Task #19: reports generation completed (path=/home/codka/Downloads/cuckoo/storage/analyses/19)
2018-01-01 12:03:14,750 [lib.cuckoo.core.scheduler] INFO: Task #19: analysis procedure completed

**Comment by doomedraven** [comment]
is snapshot in running state?

**Comment by yahyeabukar** [comment]
saved state. means i took snapshot and powered off then restored as saved state @doomedraven  i spent a lot of time but i couldn't 
thanks

**Comment by yahyeabukar** [comment]
i tried in a running state but i got this error
2018-01-01 15:16:10,964 [lib.cuckoo.core.scheduler] INFO: Using "virtualbox" as machine manager
2018-01-01 15:17:23,367 [root] CRITICAL: CuckooCriticalError: Please update your configuration. Unable to shut 'xp' down or find the machine in its proper state: Timeout hit while for machine xp to change status

**Comment by doomedraven** [comment]
it should be in running state, well if that in running state and you getting that error, no idea what is wrong, maybe recheck configuration

**Comment by yahyeabukar** [comment]
i know you are busy, please can you check my configurations in one  minute . i changed vm name winxp thanks 
virtualbox.conf

[virtualbox]
# Specify which VirtualBox mode you want to run your machines on.
# Can be "gui", "sdl" or "headless". Refer to VirtualBox's official
# documentation to understand the differences.
mode = gui

# Path to the local installation of the VBoxManage utility.
path = /usr/bin/VBoxManage
# If you are running Cuckoo on Mac OS X you have to change the path as follows:
# path = /Applications/VirtualBox.app/Contents/MacOS/VBoxManage

# Default network interface.
interface = vboxnet0

# Specify a comma-separated list of available machines to be used. For each
# specified ID you have to define a dedicated section containing the details
# on the respective machine. (E.g. cuckoo1,cuckoo2,cuckoo3)
machines = winxp

[winxp]
# Specify the label name of the current machine as specified in your
# VirtualBox configuration.
label = winxp

# Specify the operating system platform used by current machine
# [windows/darwin/linux].
platform = windows

# Specify the IP address of the current virtual machine. Make sure that the
# IP address is valid and that the host machine is able to reach it. If not,
# the analysis will fail.
ip = 192.168.56.101

# (Optional) Specify the snapshot name to use. If you do not specify a snapshot
# name, the VirtualBox MachineManager will use the current snapshot.
# Example (Snapshot1 is the snapshot name):
snapshot = Snapshot 

# (Optional) Specify the name of the network interface that should be used
# when dumping network traffic from this machine with tcpdump. If specified,
# overrides the default interface specified in auxiliary.conf
# Example (vboxnet0 is the interface name):
interface = vboxnet0

# (Optional) Specify the IP of the Result Server, as your virtual machine sees it.
# The Result Server will always bind to the address and port specified in cuckoo.conf,
# however you could set up your virtual network to use NAT/PAT, so you can specify here 
# the IP address for the Result Server as your machine sees it. If you don't specify an
# address here, the machine will use the default value from cuckoo.conf.
# NOTE: if you set this option you have to set result server IP to 0.0.0.0 in cuckoo.conf.
# Example:
# resultserver_ip = 192.168.56.1

# (Optional) Specify the port for the Result Server, as your virtual machine sees it.
# The Result Server will always bind to the address and port specified in cuckoo.conf,
# however you could set up your virtual network to use NAT/PAT, so you can specify here
# the port for the Result Server as your machine sees it. If you don't specify a port
# here, the machine will use the default value from cuckoo.conf.
# Example:
# resultserver_port = 2042

# (Optional) Set your own tags. These are comma separated and help to identify
# specific VMs. You can run samples on VMs with tag you require.
# tags = windows_xp_sp3,32_bit,acrobat_reader_6

[honeyd]
# For more information on this VM please refer to the "services" section of
# the conf/auxiliary.conf configuration file. This machine is a bit special
# in the way that its used as an additional VM for an analysis.
# *NOTE* that if this functionality is used, the VM should be registered in
# the "machines" list in the beginning of this file.
label = honeyd
platform = linux
ip = 192.168.56.1
# The tags should at least contain "service" and the name of this service.
# This way the services auxiliary module knows how to find this particular VM.
tags = service, honeyd
# Not all services actually have a Cuckoo Agent running in the VM, for those
# services one can specify the "noagent" option so Cuckoo will just wait until
# the end of the analysis instead of trying to connect to the non-existing
# Cuckoo Agent. We can't really intercept any inter-VM communication from the
# host / gateway so in order to dump traffic between VMs we have to use a
# different network dumping approach. For this machine we use the "nictrace"
# functionality from VirtualBox (which is basically their internal tcpdump)
# and thus properly dumps inter-VM traffic.
options = nictrace noagent

**Comment by yahyeabukar** [comment]
cuckoo.conf

[cuckoo]
# Enable or disable startup version check. When enabled, Cuckoo will connect
# to a remote location to verify whether the running version is the latest
# one available.
version_check = off

# If turned on, Cuckoo will delete the original file after its analysis
# has been completed.
delete_original = off

# If turned on, Cuckoo will delete the copy of the original file in the
# local binaries repository after the analysis has finished. (On *nix this
# will also invalidate the file called "binary" in each analysis directory,
# as this is a symlink.)
delete_bin_copy = off

# Specify the name of the machinery module to use, this module will
# define the interaction between Cuckoo and your virtualization software
# of choice.
machinery = virtualbox

# Enable creation of memory dump of the analysis machine before shutting
# down. Even if turned off, this functionality can also be enabled at
# submission. Currently available for: VirtualBox and libvirt modules (KVM).
memory_dump = on

# When the timeout of an analysis is hit, the VM is just killed by default.
# For some long-running setups it might be interesting to terminate the
# monitored processes before killing the VM so that connections are closed.
terminate_processes = off

# Enable automatically re-schedule of "broken" tasks each startup.
# Each task found in status "processing" is re-queued for analysis.
reschedule = off

# Enable processing of results within the main cuckoo process.
# This is the default behavior but can be switched off for setups that
# require high stability and process the results in a separate task.
process_results = on

# Limit the amount of analysis jobs a Cuckoo process goes through.
# This can be used together with a watchdog to mitigate risk of memory leaks.
max_analysis_count = 0

# Limit the number of concurrently executing analysis machines.
# This may be useful on systems with limited resources.
# Set to 0 to disable any limits.
max_machines_count = 0

# Limit the amount of VMs that are allowed to start in parallel. Generally
# speaking starting the VMs is one of the more CPU intensive parts of the
# actual analysis. This option tries to avoid maxing out the CPU completely.
max_vmstartup_count = 10

# Minimum amount of free space (in MB) available before starting a new task. 
# This tries to avoid failing an analysis because the reports can't be written 
# due out-of-diskspace errors. Setting this value to 0 disables the check.
# (Note: this feature is currently not supported under Windows.)
freespace = 64

# Temporary directory containing the files uploaded through Cuckoo interfaces
# (api.py and Django web interface).
tmppath = /tmp

# Path to the unix socket for running root commands.
rooter = /tmp/cuckoo-rooter

[routing]
# Default network routing mode; "none", "internet", or "vpn_name".
# In none mode we don't do any special routing - the VM doesn't have any
# network access (this has been the default actually for quite a while).
# In internet mode by default all the VMs will be routed through the network
# interface configured below (the "dirty line").
# And in VPN mode by default the VMs will be routed through the VPN identified
# by the given name of the VPN (as per vpn.conf).
# Note that just like enabling VPN configuration setting this option to
# anything other than "none" requires one to run utils/rooter.py as root next
# to the Cuckoo instance (as it's required for setting up the routing).
route = none

# Network interface that allows a VM to connect to the entire internet, the
# "dirty line" so to say. Note that, just like with the VPNs, this will allow
# malicious traffic through your network. So think twice before enabling it.
# (For example, to route all VMs through eth0 by default: "internet = eth0").
internet = none

# Routing table name/id for "dirty line" interface. If "dirty line" is
# also default gateway in the system you can leave "main" value. Otherwise add
# new routing table by adding "<id> <name>" line to /etc/iproute2/rt_tables
# (e.g., "200 eth0"). ID and name must be unique across the system (refer to
# /etc/iproute2/rt_tables for existing names and IDs).
rt_table = main

# To route traffic through multiple network interfaces Cuckoo uses 
# Policy Routing with separate routing table for each output interface
# (VPN or "dirty line"). If this option is enabled Cuckoo on start will try 
# to automatically initialise routing tables by copying routing entries from 
# main routing table to the new routing tables. Depending on your network/vpn 
# configuration this might not be sufficient. In such case you would need to 
# initialise routing tables manually. Note that enabling this option won't
# affect main routing table.
auto_rt = yes

[resultserver]
# The Result Server is used to receive in real time the behavioral logs
# produced by the analyzer.
# Specify the IP address of the host. The analysis machines should be able
# to contact the host through such address, so make sure it's valid.
# NOTE: if you set resultserver IP to 0.0.0.0 you have to set the option
# `resultserver_ip` for all your virtual machines in machinery configuration.
ip = 192.168.56.1

# Specify a port number to bind the result server on.
port = 2042

# Force the port chosen above, don't try another one (we can select another
# port dynamically if we can not bind this one, but that is not an option
# in some setups)
force_port = no

# Maximum size of uploaded files from VM (screenshots, dropped files, log)
# The value is expressed in bytes, by default 10Mb.
upload_max_size = 10485760

[processing]
# Set the maximum size of analyses generated files to process. This is used
# to avoid the processing of big files which may take a lot of processing
# time. The value is expressed in bytes, by default 100Mb.
analysis_size_limit = 104857600

# Enable or disable DNS lookups.
resolve_dns = on

# Enable PCAP sorting, needed for the connection content view in the web interface.
sort_pcap = on

[database]
# Specify the database connection string.
# NOTE: If you are using a custom database (different from sqlite), you have to
# use utf-8 encoding when issuing the SQL database creation statement.
# Examples, see documentation for more:
# sqlite:///foo.db
# postgresql://foo:bar@localhost:5432/mydatabase
# mysql://foo:bar@localhost/mydatabase
# If empty, default is a SQLite in db/cuckoo.db.
connection =

# Database connection timeout in seconds.
# If empty, default is set to 60 seconds.
timeout =

[timeouts]
# Set the default analysis timeout expressed in seconds. This value will be
# used to define after how many seconds the analysis will terminate unless
# otherwise specified at submission.
default = 120

# Set the critical timeout expressed in (relative!) seconds. It will be added
# to the default timeout above and after this timeout is hit
# Cuckoo will consider the analysis failed and it will shutdown the machine
# no matter what. When this happens the analysis results will most likely
# be lost.
critical = 60

# Maximum time to wait for virtual machine status change. For example when
# shutting down a vm. Default is 60 seconds.
vm_state = 60

**Comment by yahyeabukar** [comment]
sorry for the disturbing @doomedraven @jbremer

**Comment by doomedraven** [comment]
use code escape when posting something liek that it will be more easier to read, the conf looks fine, are you able to restore snapshot by yourself?

**Comment by SparkyNZL** [comment]
and a screen shot if Virtualbox to reassue us that when you snap shot the machine you did it in a working state (eg not shutting the VM down and then shapshotting)

**Comment by yahyeabukar** [comment]
@SparkyNZL sorry i did not get it , can you repeat it clearly .
@jbremer pleasse can you have any recommendation about this error thanks

**Comment by yahyeabukar** [comment]
@doomedraven i restored snapshot manually, but i think the problem is when i clicked on Unity Dash 3 different virtual box icon appear: A virtual box that i installed Windows and other 2 empty Virtual-box appear. is it possible these VM (3 VM icons) to cause the problem?

@espenfjo @heipei @jbremer

**Comment by doomedraven** [comment]
yup, uninstall vbox completely and install it, and add the existing vm

**Comment by yahyeabukar** [comment]
thanks dear @doomedraven i did it many times , i s there is any way to recognize which VM is correct one ? although i used   sudo virtualbox command to start it

**Comment by doomedraven** [comment]
if you used sudo, that will generate you a permission problem later with cuckoo. correct vm is which name you specified to it in cuckoo conf is under label

**Comment by yahyeabukar** [comment]
[SOLVED]
Thanks @doomedraven for your help. after one week suffering , i just found the solution which is : 
1) I started virtualbox without using sudo command.
2) I used $ python cuckoo.py to start cuckoo and i submitted the file  like :  $  python submit.py /home/codka/FakeIntel.exe
3) I named my virtualbox  as cuckoo1 and my snapshot : snapshot1 without space. 
i installed virtualbox-5.2 and windowsxp SP3.
My experience : Using Ubuntu 16.04 there is more than one virtualbox icon installed.  A virtual box that i installed Windows and other 2 empty Virtual-box appear in the Unity Dash.
: )
